update capture groups creation to match design

unclejustin opened the pull request:
Closes: https://github.com/sourcegraph/sourcegraph/issues/29060

Figma design: https://www.figma.com/file/qomcHxvHjrwfS3We6NVHGh/Automatically-generated-data-series-%5Breview%5D?node-id=1809%3A28970

vovakulikov commented:
@justin I fixed a few visual items and added some links there. Take a look

vovakulikov commented on client/web/src/enterprise/insights/pages/insights/creation/capture-group/components/CaptureGoupCreationForm.tsx:
```diff
@@ -101,21 +102,33 @@ export const CaptureGroupCreationForm: React.FunctionComponent<CaptureGroupCreat
                     />
 
-                    <SearchQueryChecks checks={searchQueryValidator(query.input.value)} />
+                    <SearchQueryChecks checks={searchQueryValidator(query.input.value, query.meta.touched)} />
```
Nice catch for the edit page

vovakulikov reviewed (approved):
LGTM. I added a few fixes there, take a look

vovakulikov commented:
Also it seems it closes part of https://github.com/sourcegraph/sourcegraph/issues/28878

vovakulikov commented on client/web/src/enterprise/insights/pages/insights/creation/capture-group/components/CaptureGoupCreationForm.tsx:
```diff
@@ -101,21 +102,33 @@ export const CaptureGroupCreationForm: React.FunctionComponent<CaptureGroupCreat
+                    <>
+                        Generated dynamically for each unique value from the regular expression capture group.{' '}
+                        <a
+                            href="https://docs.sourcegraph.com/code_insights/explanations/automatically_generated_data_series"
```
We need to check this link with Joel

unclejustin replied:
@Joelkw ☝️

Joelkw replied:
TY! https://github.com/sourcegraph/sourcegraph/pull/29136/files

AlicjaSuska reviewed (commented):
Thank you @unclejustin!

About the changes:
- circles in the checklist are very big. Please, let's align the sizes with the [code monitoring](https://sourcegraph.com/code-monitoring/new) creation UI
- the first circle's proportions are distorted, it's 'squeezed' from the top
- Please, see the logic about checked and unchecked items text color in [Figma](https://www.figma.com/file/qomcHxvHjrwfS3We6NVHGh/Automatically-generated-data-series-%5Breview%5D?node-id=1810%3A34861). Checked items should use text-muted color, unchecked (circles) or errors (cross) should use body color. In the current implementation in the light theme color doesn't change at all, in the dark theme it seems to be implemented in the opposite way to the design.
<img width="616" alt="Screenshot 2021-12-16 at 19 25 17" src="https://user-images.githubusercontent.com/20326070/146428965-70aafb5c-95a9-4b61-851b-fba4724a7769.png">
- styles and color used in the 'name and color...' banner are a little off. Bolder text shouldn't be bold but medium. Also, it would be ideal if we can fit text in one line as it is in Figma. I'm a little confused as to why the whole card, and this element in consequence, are way narrower than in the designs. Also, could you please change border-radius to 3px and add a border (regular border color). Radius was there in the designs, border got lost on some screens, my bad. (please note that this point is not a critical change, if there is no time to address it now, it can be done later)

General notes about the creation UI (not directly connected to this PR):
- I've noticed that the placeholder text in the repositories field says 'example: github.com.sourcegraph/sourcegraph. We needed the example before autocomplete was implemented. I think it's now safe to change to the placeholder that is in designs: 'Add or search for repositories'. @Joelkw FYI, please let me know if you have anything against it
- we should start using breader (header+bredcrums) components for the creation UI header. Please, see the [design](https://www.figma.com/file/qomcHxvHjrwfS3We6NVHGh/Automatically-generated-data-series-%5Breview%5D?node-id=1810%3A34861) and how [code monitoring](https://sourcegraph.com/code-monitoring/new) uses it.
- on second thought, I think we don't need a placeholder in the insight name. It feels a little odd. Also, code monitoring doesn't use the placeholder in the analogical case. Let's just leave the text box empty.

Joelkw commented:
> I've noticed that the placeholder text in the repositories field says 'example: github.com.sourcegraph/sourcegraph. We needed the example before autocomplete was implemented. I think it's now safe to change to the placeholder that is in designs: 'Add or search for repositories'. @Joelkw FYI, please let me know if you have anything against it

@AlicjaSuska I would normally agree (and it's not a huge deal either way if you feel strongly) except that: 

1. Our search isn't ranked/doesn't always return what you'd expect, so it's actually useful to know "oh, I should start by typing the code host url, then the repo path." If you just start by typing in "sourcegraph" you often get "sourcegraph-handbook" etc, and even if you type "sourcegraph/sourcegraph." It seems like telling someone to include the code host slug is helpful, especially if they give up on the search and type it in manually? 
2. If you look at the screenshot, all our placeholder texts are of the form `example: [example]` and I like the consistent, personally. 
![image](https://user-images.githubusercontent.com/11967660/146444115-67fbbb46-cd89-4d11-9525-e325295ebc66.png)

unclejustin commented:
Thanks @AlicjaSuska ! Please check once more time.

FWIW as an older person with presbyopia I cannot see the border around the name and color box. Also, that font-weight is so subtle as to be invisible to me. I don't think the border matters so much, but if the font-weight difference is supposed to call out and have greater meaning to the words "name" and "color" they don't for me because I literally can't see them.

AlicjaSuska commented:
@unclejustin ok, fair enough - let's use the bold font as it is now. About the border for 'name and color' box - let's add it for consistency with other places where this type of element is being used in our platform.

AlicjaSuska commented:
@unclejustin thank you for making changes.

- I've noticed that first checklist item ('Contains a properly formatted regular expression') is still on the list. It should be removed.
- I'm not sure if it's only in storybook but when I was checking the validation, list items remained checked even when I removed the search query. I've recorded the [loom](https://www.loom.com/share/6799814da5c24f289933aa6a0fbaa41c) to show what I mean.

AlicjaSuska commented:
About @Joelkw  comment regarding the placeholder text in the repository field. I think that keeping the placeholder doesn't solve our issues given that it disappears when user starts typing and our helper text below the box is not visible once the suggestion dropdown is triggered. Even if someone deletes everything from the text field, the field goes to the error state about 'required field'.

My proposition: move help text and the repo example above the field. Use helper text below the text field only for error/success messages. Placing text above the field makes it possible for people to take a look at any time when they are editing. [See in Figma](https://www.figma.com/file/qomcHxvHjrwfS3We6NVHGh/Automatically-generated-data-series-%5Breview%5D?node-id=1938%3A34437)

@Joelkw @vovakulikov @unclejustin

AlicjaSuska commented:
@unclejustin one more detail: in the regex tooltip, the text should end with a dot, not the quotation mark (') Thank you!

vovakulikov commented:
@AlicjaSuska @unclejustin I wrote you both in DM but duplicate this here as well for transparency. I think even if we're not able to finish all points around validation logic here perhaps we still want to merge this as it is (with proper copy and URLs) and finish all remaining work later in a separate PR.

unclejustin commented:
> @unclejustin thank you for making changes.
> 
> * I've noticed that first checklist item ('Contains a properly formatted regular expression') is still on the list. It should be removed.
> * I'm not sure if it's only in storybook but when I was checking the validation, list items remained checked even when I removed the search query. I've recorded the [loom](https://www.loom.com/share/6799814da5c24f289933aa6a0fbaa41c) to show what I mean.

I see. I was interpreting the request here to infer three states for the checklist items. Good, bad, and untouched. Once anything is entered in the query field it is then "touched" and there for all checks are either good or bad. An empty box is there for "all good" since it doesn't technically fail any of the checks 😅 

I've updated this to reset all of the checks if the query field is empty which feels MUCH better. Thanks for pointing this out!
